umake/frameworks/ide.py

```diff
@@ -755,3 +755,86 @@ def post_install(self):
                         exec=os.path.join(self.install_path, "Code"),
                         comment=_("Visual Studio focused on modern web and cloud"),
                         categories="Development;IDE;"))
+
+
+class SpringToolsSuite(umake.frameworks.baseinstaller.BaseInstaller):
+    def __init__(self, category):
+        super().__init__(name="Spring Tools Suite",
+                         description=_("Spring Tools Suite IDE"),
+                         download_page="https://spring.io/tools/sts/all",
+                         dir_to_decompress_in_tarball='sts-bundle/sts-*',
+                         desktop_filename='STS.desktop',
+                         category=category, only_on_archs=['i386', 'amd64'],
+                         packages_requirements=['openjdk-7-jdk'],
```

Comment:
add required_files_path=["STS" ],  in here and remove the "self.exec_path = ..." (it is now created automatically)